src/base/bittorrent/torrent.h

```diff
@@ -84,6 +86,18 @@ namespace BitTorrent
         Error
     };
 
+    class CustomNaturalLessThan
+    {
+    public:
+        bool operator()(const QString &left, const QString &right) const;
+
+    private:
+        Utils::Compare::NaturalCompare<Qt::CaseInsensitive> m_compare;
+        Utils::Compare::NaturalCompare<Qt::CaseSensitive> m_subCompare;
+    };
+
+    using TagsSet = OrderedSet<QString, CustomNaturalLessThan>;
```

Comment:
I would prefer to see `TagSet` (like `QString` in `QStringList`).

Reply:
Also I would put `TagSet` and `TagLessThan` in `tagset.h/cpp` to avoid inclusion of entire `torrent.h` where it is unnecessary.

Reply:
>Also I would put TagSet and TagLessThan in tagset.h/cpp to avoid inclusion of entire torrent.h where it is unnecessary.

src/base/tagset.h or src/base/bittorrent/tagset.h?

Reply:
Since "tag set" has no dependency from BitTorrent `src/base/tagset.h` would go.

Reply:
Done.